### PR TITLE
feat(generativeai): Add new sample for text generation

### PR DESCRIPTION
**Pull request by msampathkumar**
## Description

feat(generativeai): Add new sample for text generation(Youtube, Web support, Controlled Generations)

Fixes [b/369301403](http://b/369301403)

> This is a cleaner version of PR - https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12625

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by msampathkumar** [comment]
Update: 3.11 tests failed due service 503 error not due to bugs

File: generative_ai/audio/__init__.py
Comment (rsamborski) [inline generative_ai/audio/__init__.py:1]: Q: do we need this file?

File: generative_ai/text_generation/gemini_describe_http_image_example.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def generate_content() -> str:
+    # [START generativeaionvertexai_gemini_describe_http_image]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO (developer): update project id
```
Comment (rsamborski) [inline generative_ai/text_generation/gemini_describe_http_image_example.py:24]: nit: can you use this format: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/gemma2/gemma2_predict_gpu.py#L31-L32

File: generative_ai/text_generation/gemini_describe_http_pdf_example.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def generate_content() -> str:
+    # [START generativeaionvertexai_gemini_describe_http_pdf]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO (developer): update project id
```
Comment (rsamborski) [inline generative_ai/text_generation/gemini_describe_http_pdf_example.py:24]: nit: can you use this format: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/gemma2/gemma2_predict_gpu.py#L31-L32

File: generative_ai/text_generation/gemini_describe_http_pdf_example.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    model = GenerativeModel("gemini-1.5-flash-002")
+
+    contents = [
+        # Text prompt
+        "Summarise this file",
+        # Example video ad for Pixel 8
```
Comment (rsamborski) [inline generative_ai/text_generation/gemini_describe_http_pdf_example.py:32]: fix: The linked file is a whitepaper on trasnformers (original one), not an ad for Pixel 8

File: generative_ai/text_generation/test_text_generation_examples.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import gemini_describe_http_image_example
+import gemini_describe_http_pdf_example
+
+
+def test_gemini_describe_http_image_example() -> None:
+    text = gemini_describe_http_image_example.generate_content()
+    assert len(text) > 0
```
Comment (rsamborski) [inline generative_ai/text_generation/test_text_generation_examples.py:20]: nit: this is a very imprecise test. Can we at least check for some part of the response which shouldn't change (i.e. puppy, dog, etc.)?

File: generative_ai/text_generation/test_text_generation_examples.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def test_gemini_describe_http_pdf_example() -> None:
+    text = gemini_describe_http_pdf_example.generate_content()
+    assert len(text) > 0
```
Comment (rsamborski) [inline generative_ai/text_generation/test_text_generation_examples.py:25]: nit: ditto on the imprecise test. Maybe look for transformer in the response?

File: generative_ai/video/gemini_describe_http_video_example.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def generate_content() -> str:
+    # [START generativeaionvertexai_gemini_describe_http_video]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO (developer): update project id
```
Comment (rsamborski) [inline generative_ai/video/gemini_describe_http_video_example.py:24]: nit: ditto on the comment format

File: generative_ai/video/gemini_youtube_video_key_moments_example.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def generate_content() -> str:
+    # [START generativeaionvertexai_gemini_youtube_video_key_moments]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO (developer): update project id
```
Comment (rsamborski) [inline generative_ai/video/gemini_youtube_video_key_moments_example.py:24]: nit: ditto on the comment format

File: generative_ai/video/gemini_youtube_video_summarization_example.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def generate_content() -> str:
+    # [START generativeaionvertexai_gemini_youtube_video_summarization]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO (developer): update project id
```
Comment (rsamborski) [inline generative_ai/video/gemini_youtube_video_summarization_example.py:24]: nit: ditto on the comment format

File: generative_ai/video/gemini_youtube_video_summarization_example.py
Comment (rsamborski) [inline generative_ai/video/gemini_youtube_video_summarization_example.py:1]: Comment: This sample is practically identical to the two previous ones, not sure if we need it, but I leave the decision to you.

File: generative_ai/video/test_video_examples.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import gemini_describe_http_video_example
+import gemini_youtube_video_key_moments_example
+import gemini_youtube_video_summarization_example
+
+
+def test_gemini_describe_http_video_example() -> None:
+    text = gemini_describe_http_video_example.generate_content()
+    assert len(text) > 0
```
Comment (rsamborski) [inline generative_ai/video/test_video_examples.py:21]: nit: ditto on imprecise testing

File: generative_ai/video/test_video_examples.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def test_gemini_youtube_video_key_moments_example() -> None:
+    text = gemini_youtube_video_key_moments_example.generate_content()
+    assert len(text) > 0
```
Comment (rsamborski) [inline generative_ai/video/test_video_examples.py:26]: nit: ditto on imprecise testing

File: generative_ai/video/test_video_examples.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def test_gemini_youtube_video_summarization_example() -> None:
+    text = gemini_youtube_video_summarization_example.generate_content()
+    assert len(text) > 0
```
Comment (rsamborski) [inline generative_ai/video/test_video_examples.py:31]: nit: ditto on imprecise testing

**Review by rsamborski — APPROVED** [review]
Approved with couple comments.

**Comment by msampathkumar** [comment]
Error: "Unable to submit request because the service is temporarily out of capacity. Try again later."

**Comment by msampathkumar** [comment]
Error: `InternalServerError: 500 Internal error encountered.`

**Comment by msampathkumar** [comment]
3 tests have failed with `InternalServerError: 500 Internal error encountered.`

**Comment by msampathkumar** [comment]
Reaching out to my colleagues for additional help on this `500` error

File: generative_ai/text_generation/gemini_describe_http_image_example.py (hunk 1 above)
Comment (msampathkumar) [reply generative_ai/text_generation/gemini_describe_http_image_example.py:24]: Keeping this edit intentionally.

File: generative_ai/text_generation/test_text_generation_examples.py (hunk 4 above)
Comment (msampathkumar) [reply generative_ai/text_generation/test_text_generation_examples.py:20]: True. Goal here is only to do black box testing to verify if both the Google Cloud AI Services & SDK are working. Measuring the quality is out of scope for now.

File: generative_ai/video/gemini_youtube_video_summarization_example.py
Comment (msampathkumar) [reply generative_ai/video/gemini_youtube_video_summarization_example.py:1]: Thanks. Leaving these example for now.

**Pull request merged**